sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterOptions.java

```diff
@@ -164,10 +157,9 @@ public AzureMonitorExporterBuilder httpClient(HttpClient httpClient) {
      *
      * @param logOptions The {@link HttpLogOptions logging configuration} to use when sending and receiving requests to
      * and from the service.
-     * @return The updated {@link AzureMonitorExporterBuilder} object.
+     * @return The updated {@link AzureMonitorExporterOptions} object.
      */
-    @Override
-    public AzureMonitorExporterBuilder httpLogOptions(HttpLogOptions logOptions) {
+    public AzureMonitorExporterOptions httpLogOptions(HttpLogOptions logOptions) {
         if (frozen) {
             throw LOGGER.logExceptionAsError(new IllegalStateException(
                 "httpLogOptions cannot be changed after any of the build methods have been called"));
```

Comment:
This no longer applies for options bag as there is no `build` method anymore. So, all methods should be updated to remove this validation.

Reply:
This PR focuses on the modification of the user API. It already applies many changes. Removing the "frozen" logic, the build methods from the `AzureMonitorExportOptions` class, and other cleaning things would require changes in many places. I'd prefer that the implementation (not visible from a user API point of view) be refactored, based on the new user API, in additional PRs (we have automatic tests to catch potential mistakes). It would make the Git history easier to follow and facilitate the PR reviews.

Reply:
I might be wrong but with the new options bag the whole concept of freezing is no longer relevant - it should be ok to do 

```java
var options = new AzMonExpOptions().connectionString("foo");

// everything happens inside this method, 
// and after it's called changes to options object should not have
// any effect on exporters that's already built
AzMonExp.customize(otel, options); 

options.connectionString("bar");
AzMonExp.customize(otel, options);
```

The whole use-case is probably invalid, but you can protect from it inside `AzMonExp.cutomize` without adding any complexity into the options.

Reply:
Making build methods not visible will limit customers' option how to use our exporter. 
Certain customers don't need to use all exporters for logs, metrics and traces. They might choose to build only 1 exporter type. Similar to other languages, [they ](https://learn.microsoft.com/en-us/dotnet/api/overview/azure/monitor.opentelemetry.exporter-readme?view=azure-dotnet) have the option to build one exporter, not all. We might need to reconsider making that option available.

Reply:
If the use case is to send logs to Azure, the user could use [Azure Monitor Ingestion client library](https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/monitor/azure-monitor-ingestion#azure-monitor-ingestion-client-library-for-java).

Reply:
With the log only use case, the user can also use OpenTelemetry instrumentation libraries for Logback or Log4j 2, and inject an OpenTelemetry instance customized for Azure into them.